improver_tests/calibration/rainforests_calibration/test_ApplyRainForestsCalibrationLightGBM.py

```diff
@@ -208,7 +208,37 @@ def test__prepare_features_dataframe(ensemble_features):
     with pytest.raises(ValueError):
         ApplyRainForestsCalibrationLightGBM(
             model_config_dict={}
-        )._prepare_features_dataframe(ensemble_features)
+        )._prepare_features_array(ensemble_features)
+
+
+def test__evaluate_probabilities(
+    ensemble_features, dummy_lightgbm_models, error_threshold_cube
+):
+    """Test that _evaluate_probabilities populates error_threshold_cube.data with
+    probability data."""
+    plugin = ApplyRainForestsCalibrationLightGBM(model_config_dict={})
+    plugin.tree_models, plugin.error_thresholds = dummy_lightgbm_models
+    input_dataset, feature_variables = plugin._prepare_features_array(ensemble_features)
+    precipitation_ind = feature_variables.index("lwe_thickness_of_precipitation_amount")
+    precip_forecast = input_dataset[:, precipitation_ind]
+    data_before = error_threshold_cube.data.copy()
+    plugin._evaluate_probabilities(
+        precip_forecast, input_dataset, error_threshold_cube.data
+    )
+    diff = error_threshold_cube.data - data_before
+    # check each error threshold has been populated
+    assert np.all(np.any(diff != 0, axis=0))
+    # check data is between 0 and 1
+    assert np.all(error_threshold_cube.data >= 0)
+    assert np.all(error_threshold_cube.data <= 1)
+    # check data is 1 where forecast + error < 0
+    precip_forecast_reshaped = np.reshape(
+        precip_forecast, error_threshold_cube.data.shape[1:]
+    )
+    for i, t in enumerate(plugin.error_thresholds):
+        invalid_error = precip_forecast_reshaped + t < 0
+        print(precip_forecast_reshaped[invalid_error] + t)
```

Comment:
I don't think the print statement should be here.

Reply:
Fixed.